### ISSUE TITLE
pydantic verison should be limited to <2.0.0

### ISSUE THREAD

**Issue by gshimansky** [conda-forge/deepspeed-feedstock#23]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

Currently conda installs 2.3.0 version of pydantic as a deepspeed dependency which is incompatible with deepspeed according to https://github.com/microsoft/DeepSpeed/issues/3963#issuecomment-1636033465
Please update conda build to reflect this requirement.

### Installed packages

```shell
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    intel
_openmp_mutex             4.5                       2_gnu    intel
accimage                  0.2.0            py39h1378c84_5    conda-forge
attrs                     21.4.0             pyhd3eb1b0_0    intel
backports.zoneinfo        0.2.1            py39hf3d152e_7    conda-forge
brotli                    1.0.9                h166bdaf_8    intel
brotli-bin                1.0.9                h166bdaf_8    intel
bzip2                     1.0.8                hb9a14ef_9    intel
c-ares                    1.19.1               hd590300_0    intel
ca-certificates           2023.5.7             hbcca054_0    intel
certifi                   2023.5.7           pyhd8ed1ab_0    intel
cffi                      1.15.1           py39he91dace_3    intel
charset-normalizer        3.1.0              pyhd8ed1ab_0    intel
click                     8.0.4            py39h06a4308_0    intel
cmake                     3.27.5               hcfe8598_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    intel
contextlib2               0.5.5                      py_2    intel
cpuonly                   1.0                           0    intel
cycler                    0.11.0             pyhd3eb1b0_0    intel
cython                    0.29.35          py39h3d6467e_0    intel
daal4py                   2023.2.1          py39_intel_32    intel
dal                       2023.2.1               intel_32    intel
dataclasses               0.8                pyh6d0b6a4_7    intel
deepspeed                 0.9.5           cpu_py39hdac8457_0    conda-forge
deprecated                1.2.14             pyh1a96a4e_0    conda-forge
dpcpp-cpp-rt              2023.2.0            intel_49495    intel
dpcpp_cpp_rt              2023.2.0            intel_49495    intel
dpctl                     0.14.5          py39he78b74f_24    intel
dpnp                      0.12.1          py39hd01b3e2_157    intel
exceptiongroup            1.1.3              pyhd8ed1ab_0    conda-forge
filelock                  3.6.0              pyhd3eb1b0_0    intel
fonttools                 4.25.0             pyhd3eb1b0_0    intel
fortran_rt                2023.2.0            intel_49495    intel
freetype                  2.12.1               hb267b13_2    intel
fsspec                    2023.9.1                 pypi_0    pypi
future                    0.18.3             pyhd8ed1ab_0    intel
hjson-py                  3.1.0              pyhd8ed1ab_0    conda-forge
huggingface-hub           0.17.2                   pypi_0    pypi
hypothesis                6.86.2             pyha770c72_0    conda-forge
icc_rt                    2023.2.0            intel_49495    intel
icu                       72.1                 hcb278e6_0    intel
idna                      3.4                pyhd8ed1ab_0    intel
impi_rt                   2021.10.0           intel_49371    intel
intel-aikit-pytorch-gpu   2023.2.0                 py39_0    intel
intel-cmplr-lib-rt        2023.2.0            intel_49495    intel
intel-cmplr-lic-rt        2023.2.0            intel_49495    intel
intel-extension-for-pytorch 2.0.110              py39_xpu_0    intel
intel-fortran-rt          2023.2.0            intel_49495    intel
intel-ipp                 2019.1.144           h711154d_3    conda-forge
intel-opencl-rt           2023.2.0            intel_49495    intel
intel-openmp              2023.2.0            intel_49495    intel
intelpython               2023.2.0                      0    intel
jemalloc                  5.3.0                hcb278e6_0    conda-forge
jinja2                    3.0.1              pyhd3eb1b0_0    intel
joblib                    1.2.0              pyh3f38642_0    intel
keyutils                  1.6.1                h166bdaf_0    conda-forge
kiwisolver                1.4.4            py39hf939315_1    intel
krb5                      1.21.2               h659d440_0    conda-forge
lark-parser               0.9.0              pyh9f0ad1d_0    intel
lcms2                     2.15                 h7f713cb_2    conda-forge
lerc                      4.0.0                h27087fc_0    conda-forge
level-zero                1.11.0               h00ab1b0_0    intel
libabseil                 20230802.1      cxx17_h59595ed_0    conda-forge
libaio                    0.3.113              h166bdaf_0    conda-forge
libbrotlicommon           1.0.9                h166bdaf_8    intel
libbrotlidec              1.0.9                h166bdaf_8    intel
libbrotlienc              1.0.9                h166bdaf_8    intel
libcurl                   8.3.0                hca28451_0    conda-forge
libdeflate                1.19                 hd590300_0    conda-forge
libedit                   3.1.20210910         h7f8727e_0    intel
libev                     4.33                 h7f8727e_1    intel
libexpat                  2.5.0                hcb278e6_1    conda-forge
libffi                    3.4.2                h7f98852_5    intel
libgcc-ng                 12.2.0              h65d4601_19    intel
libgomp                   12.2.0              h65d4601_19    intel
libiconv                  1.17                 h166bdaf_0    intel
libjemalloc               5.3.0                hcb278e6_0    conda-forge
libjpeg-turbo             2.1.5.1              hd590300_1    conda-forge
libnghttp2                1.52.0               h61bc06f_0    conda-forge
libnsl                    2.0.0                h7f98852_0    intel
libpng                    1.6.39               h5eee18b_0    intel
libprotobuf               4.23.4               hf27288f_6    conda-forge
libsqlite                 3.42.0               h2797004_0    intel
libssh2                   1.11.0               h0841786_0    conda-forge
libstdcxx-ng              12.2.0              h46fd767_19    intel
libtiff                   4.6.0                h29866fb_1    conda-forge
libuuid                   2.38.1               h0b41bf4_0    intel
libuv                     1.46.0               hd590300_0    conda-forge
libwebp-base              1.3.2                hd590300_0    conda-forge
libxcb                    1.15                 h0b41bf4_0    conda-forge
libxml2                   2.11.4               h0d562d8_0    intel
libzlib                   1.2.13               hd590300_5    intel
markupsafe                2.0.1            py39h27cfd23_0    intel
matplotlib-base           3.4.3            py39hbbc1b5f_0    intel
mkl                       2023.2.0            intel_49495    intel
mkl-devel                 2023.2.0            intel_49495    intel
mkl-dpcpp                 2023.2.0            intel_49495    intel
mkl-include               2023.2.0            intel_49495    intel
mkl-service               2.4.0           py39hae59892_35    intel
mkl-static                2023.2.0            intel_49495    intel
mkl_fft                   1.3.6           py39h173b8ae_56    intel
mkl_random                1.2.2           py39h1595b48_76    intel
mkl_umath                 0.1.1           py39hd987cd3_86    intel
mpi4py                    3.1.4            py39h618b5fa_0    intel
mpmath                    1.3.0              pyhd8ed1ab_0    conda-forge
munkres                   1.1.4                      py_0    intel
ncurses                   6.4                  hcb278e6_0    intel
networkx                  2.6.2              pyhd3eb1b0_2    intel
neural-compressor         2.2                  pyhdless_0    intel
ninja                     1.11.1                   pypi_0    pypi
numpy                     1.24.3           py39hed7eef7_0    intel
numpy-base                1.24.3           py39he88ecf9_0    intel
oneccl-bind-pt            2.0.100+cpu              pypi_0    pypi
oneccl_bind_pt            2.0.100              py39_gpu_0    intel
onnx                      1.14.1           py39h4de29b9_1    conda-forge
opencv-python-headless    4.8.0.74                 py39_0    intel
openjpeg                  2.5.0                h488ebb8_3    conda-forge
openssl                   3.1.2                hd590300_0    conda-forge
packaging                 23.1               pyhd8ed1ab_0    intel
pandas                    1.5.3            py39h6cd0baa_0    intel
pillow                    10.0.1           py39h444a776_0    conda-forge
pip                       23.1.2             pyhd8ed1ab_0    intel
platformdirs              3.6.0              pyhd8ed1ab_0    intel
pooch                     1.7.0              pyha770c72_3    intel
prettytable               0.7.2                      py_3    intel
protobuf                  4.23.4           py39h60f6b12_2    conda-forge
psutil                    5.8.0            py39h27cfd23_2    intel
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
py-cpuinfo                8.0.0              pyhd3eb1b0_0    intel
pycocotools               2.0.2            py39hce5d2b2_2    intel
pycparser                 2.21               pyhd3eb1b0_0    intel
pydantic                  1.10.12          py39hd1e30aa_1    conda-forge
pyparsing                 3.1.0              pyhd8ed1ab_0    intel
pysocks                   1.7.1              pyha2e5f31_6    intel
python                    3.9.16              hef7c979_23    intel
python-dateutil           2.8.2                    py39_2    intel
python_abi                3.9                      2_cp39    intel
pytorch                   2.0.1                py39_xpu_0    intel
pytz                      2023.3             pyhd8ed1ab_0    intel
pyyaml                    6.0              py39hb9d737c_5    intel
readline                  8.2                  h8228510_1    intel
regex                     2023.8.8                 pypi_0    pypi
requests                  2.31.0             pyhd8ed1ab_0    intel
rhash                     1.4.4                hd590300_0    conda-forge
safetensors               0.3.3                    pypi_0    pypi
schema                    0.7.4              pyh44b312d_0    intel
scikit-learn              1.2.2            py39hc236052_2    intel
scikit-learn-intelex      2023.2.1          py39_intel_32    intel
scipy                     1.10.1           py39he5fd122_0    intel
setuptools                67.7.2             pyhd8ed1ab_0    intel
six                       1.16.0             pyhd3eb1b0_1    intel
sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
sympy                     1.12               pyh04b8f61_3    conda-forge
tbb                       2021.10.0           intel_49541    intel
tbb4py                    2021.10.0       py39_intel_49541    intel
threadpoolctl             3.1.0              pyh8a188c0_0    intel
tk                        8.6.12               h1ccaba5_0    intel
tokenizers                0.13.3                   pypi_0    pypi
torchvision               0.15.2                   py39_0    intel
tornado                   6.1              py39h27cfd23_0    intel
tqdm                      4.65.0             pyhd8ed1ab_1    intel
transformers              4.33.2                   pypi_0    pypi
typing-extensions         4.6.3                hd8ed1ab_0    intel
typing_extensions         4.6.3              pyha770c72_0    intel
tzdata                    2023c                h71feb2d_0    intel
urllib3                   2.0.3              pyhd8ed1ab_0    intel
wheel                     0.40.0             pyhd8ed1ab_0    intel
wrapt                     1.13.3           py39h7f8727e_2    intel
xorg-libxau               1.0.11               hd590300_0    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xz                        5.2.8                h5eee18b_0    intel
yaml                      0.2.5                h7f98852_2    intel
zlib                      1.2.13               hd590300_5    intel
zstd                      1.5.5                hfc55251_0    conda-forge
```

### Environment info

```shell
active environment : pytorch
    active env location : /opt/conda/envs/pytorch
            shell level : 2
       user config file : /root/.condarc
 populated config files :
          conda version : 23.7.4
    conda-build version : not installed
         python version : 3.11.4.final.0
       virtual packages : __archspec=1=x86_64
                          __glibc=2.35=0
                          __linux=5.15.0=0
                          __unix=0=0
       base environment : /opt/conda  (writable)
      conda av data dir : /opt/conda/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /opt/conda/pkgs
                          /root/.conda/pkgs
       envs directories : /opt/conda/envs
                          /root/.conda/envs
               platform : linux-64
             user-agent : conda/23.7.4 requests/2.31.0 CPython/3.11.4 Linux/5.15.0-76-generic ubuntu/22.04.2 glibc/2.35
                UID:GID : 0:0
             netrc file : None
           offline mode : False
```

**Comment by loadams** [comment]
To add context from the DeepSpeed side, we are working on adding PyDantic 2+ support, but aren't there yet, and our `requirements.txt` specifies `pydantic<2.0.0`, so if possible, it would help users to not hit the error if they're using a newer pydantic version.

**Comment by hadim** [comment]
I have limited bandwidth currently, but I would be happy to review and merge a PR on the feedstock.

**Comment by weiji14** [comment]
Can probably unpin `pydantic<2.0` once the patch at https://github.com/microsoft/DeepSpeed/pull/4407 is released in `deepspeed` v0.11.2 or newer.

**Comment by loadams** [comment]
@weiji14 -  agreed, we should have a new release within the next week or so.
